packages/middleware-bucket-endpoint/src/bucketHostnameUtils.ts

```diff
@@ -0,0 +1,128 @@
+import { ARN } from "@aws-sdk/util-arn-parser";
+
+const DOMAIN_PATTERN = /^[a-z0-9][a-z0-9\.\-]{1,61}[a-z0-9]$/;
+const IP_ADDRESS_PATTERN = /(\d+\.){3}\d+/;
+const DOTS_PATTERN = /\.\./;
+export const DOT_PATTERN = /\./;
+export const S3_HOSTNAME_PATTERN = /^(.+\.)?s3[.-]([a-z0-9-]+)\./;
+const S3_US_EAST_1_ALTNAME_PATTERN = /^s3(-external-1)?\.amazonaws\.com$/;
+const AWS_PARTITION_SUFFIX = "amazonaws.com";
+
+export interface AccessPointArn extends ARN {
+  accessPointName: string;
+}
+
+export interface BucketHostnameParameters {
```

Comment:
nit: Should be shortened to BucketHostnameParams?
```suggestion
export interface BucketHostnameParams {
```

Reply:
Addressed in https://github.com/aws/aws-sdk-js-v3/pull/1441/commits/29089434b3a739716668caec43d13f979dd620c3